### DISCUSSION TITLE
next/image is mixing up images..

### DISCUSSION THREAD

**Discussion by lukethegeek** [vercel/next.js#27353] [Help]
I've encountered a rather strange issue, I'm not sure if it's something that I'm doing wrong or a bug with next/image. Perhaps someone else has encountered this?

I'm loading multiple small images as part of a gallery type thing (30 initial load, plus 30 preload for pagination), these images are fairly large on the backend so I'm relying on next/image to resize and optimise, it works great except.. sometimes the images get mixed up or duplicated..

You can see in the image below that the two hearts and the two at the bottom right are duplicated, when they all should be displaying their own image. Sometimes all the images are displayed properly, sometimes not.

![image](https://user-images.githubusercontent.com/5013661/126370413-49ab3ed9-3fee-405b-b4af-125662513f3e.png)

The funny thing is that next is actually caching and serving the wrong image - it doesn't appear to be a React hydration issue. If I access the next/image url directly, next is definitely serving the wrong image for ones affected. However if I change a parameter like increment the quality param, the correct image is then generated and displayed for that specific quality level.

So somehow there appears to be some kind of 'cross-talk' when 30-60 images being generated and cached in one go by next/image and they are getting mixed up..

Edit:
It appears to be happening for only certain image resolutions within the srcset at random so I suppose there's a random chance for the wrong image to be served as they are resized / optimised.

Having used the responsive design developer tools to test different resolutions, I now have a gallery image shown in the footer where the logo should be. The URL is correct for that image element, but the image being served by next is incorrect.

![image](https://user-images.githubusercontent.com/5013661/126377707-91b8fcdc-5127-4a35-9b9a-672b346abb95.png)

**Comment by luizeboli** [comment]
Related to https://github.com/vercel/next.js/issues/23518
